server/src/test/java/org/opensearch/rest/action/admin/cluster/RestClusterStatsActionTests.java

```diff
@@ -0,0 +1,159 @@
+        }
+    }
+
+    public void testFromRequestWithNodeStatsMetricsFilter() {
+        Set<ClusterStatsRequest.Metric> metricsRequested = Set.of(ClusterStatsRequest.Metric.OS, ClusterStatsRequest.Metric.FS);
+        String metricParam = metricsRequested.stream().map(ClusterStatsRequest.Metric::metricName).collect(Collectors.joining(","));
+        final HashMap<String, String> params = new HashMap<>();
+        params.put("metric", metricParam);
+        final RestRequest request = new FakeRestRequest.Builder(xContentRegistry()).withPath("/_cluster/stats").withParams(params).build();
+        ClusterStatsRequest clusterStatsRequest = RestClusterStatsAction.fromRequest(request);
+        assertNotNull(clusterStatsRequest);
+        assertTrue(clusterStatsRequest.useAggregatedNodeLevelResponses());
+        assertTrue(clusterStatsRequest.computeAllMetrics());
+        assertFalse(clusterStatsRequest.requestedMetrics().isEmpty());
+        assertEquals(2, clusterStatsRequest.requestedMetrics().size());
+        for (ClusterStatsRequest.Metric metric : metricsRequested) {
+            assertTrue(clusterStatsRequest.requestedMetrics().contains(metric));
+        }
+        assertTrue(clusterStatsRequest.indicesMetrics().isEmpty());
+    }
+
+    public void testFromRequestWithIndicesStatsMetricsFilter() {
+        Set<ClusterStatsRequest.Metric> metricsRequested = Set.of(
+            ClusterStatsRequest.Metric.OS,
+            ClusterStatsRequest.Metric.FS,
+            ClusterStatsRequest.Metric.INDICES
+        );
+        Set<ClusterStatsRequest.IndexMetric> indicesMetricsRequested = Set.of(
+            ClusterStatsRequest.IndexMetric.SHARDS,
+            ClusterStatsRequest.IndexMetric.SEGMENTS
+        );
+        String metricParam = metricsRequested.stream().map(ClusterStatsRequest.Metric::metricName).collect(Collectors.joining(","));
+        String indicesMetricParam = indicesMetricsRequested.stream()
+            .map(ClusterStatsRequest.IndexMetric::metricName)
+            .collect(Collectors.joining(","));
+        final HashMap<String, String> params = new HashMap<>();
+        params.put("metric", metricParam);
+        params.put("index_metric", indicesMetricParam);
+        final RestRequest request = new FakeRestRequest.Builder(xContentRegistry()).withPath("/_cluster/stats").withParams(params).build();
+        ClusterStatsRequest clusterStatsRequest = RestClusterStatsAction.fromRequest(request);
+        assertNotNull(clusterStatsRequest);
+        assertTrue(clusterStatsRequest.useAggregatedNodeLevelResponses());
+        assertTrue(clusterStatsRequest.computeAllMetrics());
+        assertFalse(clusterStatsRequest.requestedMetrics().isEmpty());
+        assertEquals(3, clusterStatsRequest.requestedMetrics().size());
+        for (ClusterStatsRequest.Metric metric : metricsRequested) {
+            assertTrue(clusterStatsRequest.requestedMetrics().contains(metric));
+        }
```

Comment:
We can rather have local `metricsRequested ` as list and perform  List#equals comparison

Reply:
Thanks. Updated.